autoload/go/asmfmt.vim

```diff
@@ -41,7 +41,11 @@ function! go#asmfmt#Format()
 
     " Replace the current file with the temp file; then reload the buffer.
     let old_fileformat = &fileformat
+    " save old file permissions
+    let original_fperm = getfperm(expand('%'))
```

Comment:
According to the doc this also returns `-` if the user does not have the given permission. We should test if the variable is `-` below before we set it to prevent unwanted errors.